app/main.py

```diff
@@ -86,128 +87,64 @@
 }
 
 
-def battle(knightsConfig):
+class Knight:
+    def __init__(
+            self,
+            name: int, power: int, hp: int,
+            armour: dict, weapon: dict, potion: dict = None) -> None:
```

Comment:
Place each argument on a new line to improve code style and readability. There should be no arguments at the same line with closing parenthesis.